pkg/templates/aws/edges/ecr_image-ecr_repo.yaml

```diff
@@ -1,2 +1,3 @@
 source: aws:ecr_image
 target: aws:ecr_repo
+unique: one-to-one
```

Comment:
Wouldn't this make multiple repos, one per image instead of re-using the repo for multiple images? Should it be `many-to-one`?

Reply:
so currently we dont support more than 1 image per repo based on how our iac is, but we could change this and just make the repo generation unique everywhere for images via op rules instead

Reply:
What about our iac prevents hosting more than 1 image to a repo? It's fine if we're working around it, just put that (not just 'what', but also 'why') in comments here in the meantime in that case.

Reply:
its because we push it with the same name so we overwrite the image if you have multiple images we are trying to push, we would have to start tracking tags for the images we are pulling in because in compute we feed the image.imageName which is not distinguishable between tags i dont believe. We can dig into it further, but in e2e tests i deployed 2 lambdas and they both ended up using the same image

Reply:
Ah, we don't really set the name, which is odd. `${args.Repo.repositoryUrl}:` just pushes to the repo root... I'm surprised that even works. Couldn't we just throw the `Name` in there to support it? Ie, something like:
```js
pulumi.interpolate`${args.Repo.repositoryUrl}/{{.Name}}:{{ if .Tag }}${args.Tag}-{{ end }}${sha256}`
```